### PR TITLE
emotes baybee

### PR DESCRIPTION
**Pull request by Pacmandevil**
Adds several emotes.

*vomit - as suggested by this fourm thread [here](https://forums.aurorastation.org/viewtopic.php?f=18&p=82986&sid=43d315122d63b1c27241f5139fe63184#p82986) - This has cooldowns to prevent spam.

*slap - an emote nobody will figure out how to use after slapping themselves 5 times by accident.

and *snap, for recreating that scene in a musical where you walk up to someone with a switchblade while wearing a leather jacket - snapping your fingers every step while overly dramatic music plays in the backround.

### CODE REVIEW

**Comment by Arrow768** [comment]
Please make sure that checking the vomit puddle for DNA actually works.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -613,13 +649,27 @@
 					message = "buzzes."
 				playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
 				m_type = 1
+		
+		if("vomit")
+			if (isipc(src))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/emote.dm]: I'm pretty sure there's a, "hasmouth()" proc or something like that. Use that.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -1,6 +1,6 @@
+/mob/living/carbon/human/var/emotecooldown = 0
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/emote.dm]: Isn't there a specific vomit cooldown already for a human?

File: code/modules/mob/living/carbon/human/emote.dm [hunk 3]
```diff
@@ -613,13 +649,27 @@
...
+			return
+
 
 		if ("help")
 			src << {"blink, blink_r, blush, bow-(none)/mob, burp, choke, chuckle, clap, collapse, cough,
 cry, custom, deathgasp, drool, eyebrow, frown, gasp, giggle, groan, grumble, handshake, hug-(none)/mob, glare-(none)/mob,
 grin, laugh, look-(none)/mob, moan, mumble, nod, pale, point-atom, raise, salute, shake, shiver, shrug,
 sigh, signal-#1-10, smile, sneeze, sniff, snore, stare-(none)/mob, tremble, twitch, twitch_s, whimper,
-wink, yawn, swish, sway/wag, fastsway/qwag, stopsway/swag, beep, ping, buzz"}
+wink, yawn, swish, sway/wag, fastsway/qwag, stopsway/swag, beep, ping, buzz, slap, snap"}
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/emote.dm]: Either document vomit, or remove it.

Comment (Pacmandevil) [reply code/modules/mob/living/carbon/human/emote.dm]: There's other various emotes that arent documented, including *dap, though I will admit that I just forgot to add it to the list.

**Comment by skull132** [comment]
@Pacmandevil aaaaaaaaaaaaaaaaa plz do changes.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 2 above)
Comment (SoundScopes) [reply code/modules/mob/living/carbon/human/emote.dm]: It's defined in "delayed_vomit()" not "vomit()" itself.
delayed_vomit() also uses three spawn() calls which will probably bug Loh.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 1 above)
Comment (SoundScopes) [reply code/modules/mob/living/carbon/human/emote.dm]: check_has_mouth() is the call you want

File: code/modules/mob/living/carbon/human/emote.dm (hunk 2 above)
Comment (Lohikar) [reply code/modules/mob/living/carbon/human/emote.dm]: _eyetwitch_

File: code/modules/mob/living/carbon/human/emote.dm [hunk 4]
```diff
@@ -555,8 +590,8 @@
 			src.animate_tail_stop()
 
 		if("beep")
-			if (!isipc(src))
-				src << span("notice", "You're not a machine!")
+			if (!check_has_mouth(src))
```
Comment (SoundScopes) [inline code/modules/mob/living/carbon/human/emote.dm]: This one should be !isipc(src) the vomit check was the one that needed check_has_mouth(src)

Comment (Pacmandevil) [reply code/modules/mob/living/carbon/human/emote.dm]: this is why you don't code while piss-tired ree

**Pull request merged**